### ISSUE TITLE
dotCMS db locked placeholder - potential issue

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#9156]
I just had my local development instance lock up - basically nothing could get a db connection.  I took a few thread dumps - two of which are included in this log file:

https://gist.github.com/wezell/034025f179b5e796e8ac6e935753f5a8

but basically, VirtualLink db lookups were unable to process.  Seeing the ClusterThread involved got me concerned as I know we have been doing work here.

**Comment by wezell** [comment]
So it just happened to me again, when I undeployed an osgi plugin (it was a portlet with no db work)

**Comment by wezell** [comment]
And again.  No plugins running- nothing.  Threaddump

https://gist.github.com/wezell/5a5d1186e70d2e01e6588d0fee8493f5

**Comment by wezell** [comment]
DB

```
mysql> show processlist;
+----+--------+-----------------+---------+---------+------+-------+------------------+
| Id | User   | Host            | db      | Command | Time | State | Info             |
+----+--------+-----------------+---------+---------+------+-------+------------------+
| 91 | dotcms | localhost:63277 | dotcms3 | Sleep   |  262 |       | NULL             |
| 92 | dotcms | localhost:63279 | dotcms3 | Sleep   |  322 |       | NULL             |
| 93 | dotcms | localhost:63280 | dotcms3 | Sleep   |  502 |       | NULL             |
| 94 | dotcms | localhost:63300 | dotcms3 | Sleep   |  442 |       | NULL             |
| 95 | dotcms | localhost:63301 | dotcms3 | Sleep   |  382 |       | NULL             |
| 96 | dotcms | localhost:63319 | dotcms3 | Sleep   |  142 |       | NULL             |
| 97 | dotcms | localhost:63362 | dotcms3 | Sleep   |  142 |       | NULL             |
| 98 | dotcms | localhost:63372 | dotcms3 | Sleep   |  202 |       | NULL             |
| 99 | root   | localhost:63441 | dotcms3 | Query   |    0 | init  | show processlist |
+----+--------+-----------------+---------+---------+------+-------+------------------+
9 rows in set (0.00 sec)
```

**Comment by dsilvam** [comment]
@wezell I just pulled latest changes from master-3.6 and started fresh with empty db and everything got up and running ok. We can check on Tuesday what is going on

**Comment by wezell** [comment]
So I turned on db abandoned connection pool logging yesterday and and just sat back down at my computer and see this in my logs:

```
INFO  2016-05-30 20:24:44,118 com.dotmarketing.quartz.DotJobStore:info(260): Handling 5 trigger(s) that missed their scheduled fire-time.
:87)
	at com.dotmarketing.common.db.DotConnect.executeQuery(DotConnect.java:489)
	at com.dotmarketing.common.db.DotConnect.loadResult(DotConnect.java:232)
	at com.dotmarketing.common.db.DotConnect.loadObjectResults(DotConnect.java:863)
	at com.dotcms.enterprise.cluster.ServerFactoryImpl.getAllServers(ServerFactoryImpl.java:141)
	at com.dotcms.cluster.business.ServerAPIImpl.getAllServers(ServerAPIImpl.java:138)
	at com.dotcms.cluster.business.ServerAPIImpl.getInactiveServers(ServerAPIImpl.java:152)
	at com.dotmarketing.quartz.job.FreeServerFromClusterJob.execute(FreeServerFromClusterJob.java:35)
	at org.quartz.core.JobRunShell.run(JobRunShell.java:223)
	at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:549)
org.apache.tomcat.dbcp.pool2.impl.DefaultPooledObject$AbandonedObjectCreatedException: Pooled object created 2016-05-29 19:44:55 -0400 by the following code has not been returned to the pool:
	at org.apache.tomcat.dbcp.pool2.impl.DefaultPooledObject.allocate(DefaultPooledObject.java:196)
	at org.apache.tomcat.dbcp.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:449)
	at org.apache.tomcat.dbcp.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:360)
	at org.apache.tomcat.dbcp.dbcp2.PoolingDataSource.getConnection(PoolingDataSource.java:118)
	at org.apache.tomcat.dbcp.dbcp2.BasicDataSource.getConnection(BasicDataSource.java:1412)
	at com.dotmarketing.db.DbConnectionFactory.getConnection(DbConnectionFactory.java:87)
	at com.dotmarketing.common.db.DotConnect.executeQuery(DotConnect.java:489)
	at com.dotmarketing.common.db.DotConnect.loadResult(DotConnect.java:232)
	at com.dotmarketing.common.db.DotConnect.loadObjectResults(DotConnect.java:863)
	at com.dotcms.enterprise.cluster.ServerFactoryImpl.getAllServers(ServerFactoryImpl.java:141)
	at com.dotcms.cluster.business.ServerAPIImpl.getAllServers(ServerAPIImpl.java:138)
	at com.dotcms.cluster.business.ServerAPIImpl.getInactiveServers(ServerAPIImpl.java:152)
	at com.dotmarketing.quartz.job.FreeServerFromClusterJob.execute(FreeServerFromClusterJob.java:35)
	at org.quartz.core.JobRunShell.run(JobRunShell.java:223)
	at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:549)
org.apache.tomcat.dbcp.pool2.impl.DefaultPooledObject$AbandonedObjectCreatedException: Pooled object created 2016-05-30 01:22:31 -0400 by the following code has not been returned to the pool:
	at org.apache.tomcat.dbcp.pool2.impl.DefaultPooledObject.allocate(DefaultPooledObject.java:196)
	at org.apache.tomcat.dbcp.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:449)
	at org.apache.tomcat.dbcp.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:360)
	at org.apache.tomcat.dbcp.dbcp2.PoolingDataSource.getConnection(PoolingDataSource.java:118)
	at org.apache.tomcat.dbcp.dbcp2.BasicDataSource.getConnection(BasicDataSource.java:1412)
	at com.dotmarketing.db.DbConnectionFactory.getConnection(DbConnectionFactory.java:87)
	at com.dotmarketing.common.db.DotConnect.executeQuery(DotConnect.java:489)
	at com.dotmarketing.common.db.DotConnect.loadResult(DotConnect.java:232)
	at com.dotmarketing.common.db.DotConnect.loadObjectResults(DotConnect.java:863)
	at com.dotcms.enterprise.cluster.ServerFactoryImpl.getAllServers(ServerFactoryImpl.java:141)
	at com.dotcms.cluster.business.ServerAPIImpl.getAllServers(ServerAPIImpl.java:138)
	at com.dotcms.cluster.business.ServerAPIImpl.getInactiveServers(ServerAPIImpl.java:152)
	at com.dotmarketing.quartz.job.FreeServerFromClusterJob.execute(FreeServerFromClusterJob.java:35)
	at org.quartz.core.JobRunShell.run(JobRunShell.java:223)
	at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:549)
org.apache.tomcat.dbcp.pool2.impl.DefaultPooledObject$AbandonedObjectCreatedException: Pooled object created 2016-05-30 00:13:41 -0400 by the following code has not been returned to the pool:
	at org.apache.tomcat.dbcp.pool2.impl.DefaultPooledObject.allocate(DefaultPooledObject.java:196)
	at org.apache.tomcat.dbcp.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:449)
	at org.apache.tomcat.dbcp.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:360)
	at org.apache.tomcat.dbcp.dbcp2.PoolingDataSource.getConnection(PoolingDataSource.java:118)
	at org.apache.tomcat.dbcp.dbcp2.BasicDataSource.getConnection(BasicDataSource.java:1412)
	at com.dotmarketing.db.DbConnectionFactory.getConnection(DbConnectionFactory.java:87)
	at com.dotmarketing.common.db.DotConnect.executeQuery(DotConnect.java:489)
	at com.dotmarketing.common.db.DotConnect.loadResult(DotConnect.java:232)
	at com.dotmarketing.common.db.DotConnect.loadObjectResults(DotConnect.java:863)
	at com.dotcms.enterprise.cluster.ServerFactoryImpl.getAllServers(ServerFactoryImpl.java:141)
	at com.dotcms.cluster.business.ServerAPIImpl.getAllServers(ServerAPIImpl.java:138)
	at com.dotcms.cluster.business.ServerAPIImpl.getInactiveServers(ServerAPIImpl.java:152)
	at com.dotmarketing.quartz.job.FreeServerFromClusterJob.execute(FreeServerFromClusterJob.java:35)
	at org.quartz.core.JobRunShell.run(JobRunShell.java:223)
	at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:549)
org.apache.tomcat.dbcp.pool2.impl.DefaultPooledObject$AbandonedObjectCreatedException: Pooled object created 2016-05-30 03:37:03 -0400 by the following code has not been returned to the pool:
	at org.apache.tomcat.dbcp.pool2.impl.DefaultPooledObject.allocate(DefaultPooledObject.java:196)
	at org.apache.tomcat.dbcp.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:449)
	at org.apache.tomcat.dbcp.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:360)
	at org.apache.tomcat.dbcp.dbcp2.PoolingDataSource.getConnection(PoolingDataSource.java:118)
	at org.apache.tomcat.dbcp.dbcp2.BasicDataSource.getConnection(BasicDataSource.java:1412)
	at com.dotmarketing.db.DbConnectionFactory.getConnection(DbConnectionFactory.java:87)
	at com.dotmarketing.common.db.DotConnect.executeQuery(DotConnect.java:489)
	at com.dotmarketing.common.db.DotConnect.loadResult(DotConnect.java:232)
	at com.dotmarketing.common.db.DotConnect.loadObjectResults(DotConnect.java:863)
	at com.dotcms.enterprise.cluster.ServerFactoryImpl.getAllServers(ServerFactoryImpl.java:141)
	at com.dotcms.cluster.business.ServerAPIImpl.getAllServers(ServerAPIImpl.java:138)
	at com.dotcms.cluster.business.ServerAPIImpl.getInactiveServers(ServerAPIImpl.java:152)
	at com.dotmarketing.quartz.job.FreeServerFromClusterJob.execute(FreeServerFromClusterJob.java:35)
	at org.quartz.core.JobRunShell.run(JobRunShell.java:223)
	at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:549)
org.apache.tomcat.dbcp.pool2.impl.DefaultPooledObject$AbandonedObjectCreatedException: Pooled object created 2016-05-30 03:37:03 -0400 by the following code has not been returned to the pool:
	at org.apache.tomcat.dbcp.pool2.impl.DefaultPooledObject.allocate(DefaultPooledObject.java:196)
	at org.apache.tomcat.dbcp.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:449)
	at org.apache.tomcat.dbcp.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:360)
	at org.apache.tomcat.dbcp.dbcp2.PoolingDataSource.getConnection(PoolingDataSource.java:118)
	at org.apache.tomcat.dbcp.dbcp2.BasicDataSource.getConnection(BasicDataSource.java:1412)
	at com.dotmarketing.db.DbConnectionFactory.getConnection(DbConnectionFactory.java:87)
	at com.dotmarketing.db.HibernateUtil.getSession(HibernateUtil.java:638)
	at com.dotmarketing.db.HibernateUtil.setQuery(HibernateUtil.java:194)
	at com.dotmarketing.factories.InodeFactory.getInodeOfClassByCondition(InodeFactory.java:868)
	at com.dotmarketing.portlets.structure.factories.FieldFactory.getFieldByVariableName(FieldFactory.java:81)
	at com.dotcms.enterprise.priv.DashboardFactoryImpl.<init>(DashboardFactoryImpl.java:56)
	at com.dotcms.enterprise.DashboardProxy.getDashboardFactory(DashboardProxy.java:12)
	at com.dotmarketing.business.FactoryIndex.create(FactoryLocator.java:330)
	at com.dotmarketing.business.FactoryLocator.createService(FactoryLocator.java:256)
	at com.dotmarketing.business.FactoryLocator.createServiceINFO  2016-05-30 21:20:45,001 com.dotmarketing.quartz.DotJobStore:info(260): Handling 4 trigger(s) that missed their scheduled fire-time.
(FactoryLocator.java:1)
	at com.dotmarketing.business.Locator.getServiceInstance(Locator.java:41)
	at com.dotmarketing.business.FactoryLocator.getInstance(FactoryLocator.java:246)
	at com.dotmarketing.business.FactoryLocator.getDashboardFactory(FactoryLocator.java:172)
	at com.dotmarketing.portlets.dashboard.business.DashboardAPIImpl.<clinit>(DashboardAPIImpl.java:26)
	at com.dotmarketing.business.APIIndex.create(APILocator.java:496)
	at com.dotmarketing.business.APILocator.createService(APILocator.java:390)
	at com.dotmarketing.business.APILocator.createService(APILocator.java:1)
	at com.dotmarketing.business.Locator.getServiceInstance(Locator.java:41)
	at com.dotmarketing.business.APILocator.getInstance(APILocator.java:378)
	at com.dotmarketing.business.APILocator.getDashboardAPI(APILocator.java:274)
	at com.dotcms.enterprise.priv.DashboardJobImpl.<init>(DashboardJobImpl.java:19)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:422)
	at java.lang.Class.newInstance(Class.java:442)
	at org.quartz.simpl.SimpleJobFactory.newJob(SimpleJobFactory.java:55)
	at org.quartz.core.JobRunShell.initialize(JobRunShell.java:140)
	at org.quartz.core.QuartzSchedulerThread.run(QuartzSchedulerThread.java:364)
org.apache.tomcat.dbcp.pool2.impl.DefaultPooledObject$AbandonedObjectCreatedException: Pooled object created 2016-05-30 06:59:00 -0400 by the following code has not been returned to the pool:
	at org.apache.tomcat.dbcp.pool2.impl.DefaultPooledObject.allocate(DefaultPooledObject.java:196)
	at org.apache.tomcat.dbcp.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:449)
	at org.apache.tomcat.dbcp.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:360)
	at org.apache.tomcat.dbcp.dbcp2.PoolingDataSource.getConnection(PoolingDataSource.java:118)
	at org.apache.tomcat.dbcp.dbcp2.BasicDataSource.getConnection(BasicDataSource.java:1412)
	at com.dotmarketing.db.DbConnectionFactory.getConnection(DbConnectionFactory.java:87)
	at com.dotmarketing.common.db.DotConnect.executeQuery(DotConnect.java:489)
	at com.dotmarketing.common.db.DotConnect.loadResult(DotConnect.java:232)
	at com.dotmarketing.common.db.DotConnect.loadObjectResults(DotConnect.java:863)
	at com.dotcms.enterprise.cluster.ServerFactoryImpl.getAllServers(ServerFactoryImpl.java:141)
	at com.dotcms.cluster.business.ServerAPIImpl.getAllServers(ServerAPIImpl.java:138)
	at com.dotcms.cluster.business.ServerAPIImpl.getInactiveServers(ServerAPIImpl.java:152)
	at com.dotmarketing.quartz.job.FreeServerFromClusterJob.execute(FreeServerFromClusterJob.java:35)
	at org.quartz.core.JobRunShell.run(JobRunShell.java:223)
	at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:549)
org.apache.tomcat.dbcp.pool2.impl.DefaultPooledObject$AbandonedObjectCreatedException: Pooled object created 2016-05-30 06:58:43 -0400 by the following code has not been returned to the pool:
	at org.apache.tomcat.dbcp.pool2.impl.DefaultPooledObject.allocate(DefaultPooledObject.java:196)
	at org.apache.tomcat.dbcp.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:449)
	at org.apache.tomcat.dbcp.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:360)
	at org.apache.tomcat.dbcp.dbcp2.PoolingDataSource.getConnection(PoolingDataSource.java:118)
	at org.apache.tomcat.dbcp.dbcp2.BasicDataSource.getConnection(BasicDataSource.java:1412)
	at com.dotmarketing.db.DbConnectionFactory.getConnection(DbConnectionFactory.java:87)
	at com.dotmarketing.common.db.DotConnect.executeQuery(DotConnect.java:489)
	at com.dotmarketing.common.db.DotConnect.loadResult(DotConnect.java:232)
	at com.dotmarketing.common.db.DotConnect.loadObjectResults(DotConnect.java:863)
	at com.dotcms.enterprise.cluster.ServerFactoryImpl.getAllServers(ServerFactoryImpl.java:141)
	at com.dotcms.cluster.business.ServerAPIImpl.getAllServers(ServerAPIImpl.java:138)
	at com.dotcms.cluster.business.ServerAPIImpl.getInactiveServers(ServerAPIImpl.java:152)
	at com.dotmarketing.quartz.job.FreeServerFromClusterJob.execute(FreeServerFromClusterJob.java:35)
	at org.quartz.core.JobRunShell.run(JobRunShell.java:223)
	at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:549)
org.apache.tomcat.dbcp.pool2.impl.DefaultPooledObject$AbandonedObjectCreatedException: Pooled object created 2016-05-30 09:12:25 -0400 by the following code has not been returned to the pool:
	at org.apache.tomcat.dbcp.pool2.impl.DefaultPooledObject.allocate(DefaultPooledObject.java:196)
	at org.apache.tomcat.dbcp.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:449)
	at org.apache.tomcat.dbcp.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:360)
	at org.apache.tomcat.dbcp.dbcp2.PoolingDataSource.getConnection(PoolingDataSource.java:118)
	at org.apache.tomcat.dbcp.dbcp2.BasicDataSource.getConnection(BasicDataSource.java:1412)
	at com.dotmarketing.db.DbConnectionFactory.getConnection(DbConnectionFactory.java:87)
	at com.dotmarketing.common.db.DotConnect.executeQuery(DotConnect.java:489)
	at com.dotmarketing.common.db.DotConnect.loadResult(DotConnect.java:232)
	at com.dotmarketing.common.db.DotConnect.loadObjectResults(DotConnect.java:863)
	at com.dotcms.enterprise.cluster.ServerFactoryImpl.getAllServers(ServerFactoryImpl.java:141)
	at com.dotcms.cluster.business.ServerAPIImpl.getAllServers(ServerAPIImpl.java:138)
	at com.dotcms.cluster.business.ServerAPIImpl.getInactiveServers(ServerAPIImpl.java:152)
	at com.dotmarketing.quartz.job.FreeServerFromClusterJob.execute(FreeServerFromClusterJob.java:35)
	at org.quartz.core.JobRunShell.run(JobRunShell.java:223)
	at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:549)
org.apache.tomcat.dbcp.pool2.impl.DefaultPooledObject$AbandonedObjectCreatedException: Pooled object created 2016-05-30 17:02:59 -0400 by the following code has not been returned to the pool:
	at org.apache.tomcat.dbcp.pool2.impl.DefaultPooledObject.allocate(DefaultPooledObject.java:196)
	at org.apache.tomcat.dbcp.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:449)
	at org.apache.tomcat.dbcp.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:360)
	at org.apache.tomcat.dbcp.dbcp2.PoolingDataSource.getConnection(PoolingDataSource.java:118)
	at org.apache.tomcat.dbcp.dbcp2.BasicDataSource.getConnection(BasicDataSource.java:1412)
	at com.dotmarketing.db.DbConnectionFactory.getConnection(DbConnectionFactory.java:87)
	at com.dotmarketing.common.db.DotConnect.executeQuery(DotConnect.java:489)
	at com.dotmarketing.common.db.DotConnect.loadResult(DotConnect.java:232)
	at com.dotmarketing.common.db.DotConnect.loadObjectResults(DotConnect.java:863)
	at com.dotcms.enterprise.cluster.ServerFactoryImpl.getAllServers(ServerFactoryImpl.java:141)
	at com.dotcms.cluster.business.ServerAPIImpl.getAllServers(ServerAPIImpl.java:138)
	at com.dotcms.cluster.business.ServerAPIImpl.getInactiveServers(ServerAPIImpl.java:152)
	at com.dotmarketing.quartz.job.FreeServerFromClusterJob.execute(FreeServerFromClusterJob.java:35)
	at org.quartz.core.JobRunShell.run(JobRunShell.java:223)
	at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:549)
org.apache.tomcat.dbcp.pool2.impl.DefaultPooledObject$AbandonedObjectCreatedException: Pooled object created 2016-05-30 17:03:00 -0400 by the following code has not been returned to the pool:
	at org.apache.tomcat.dbcp.pool2.impl.DefaultPooledObject.allocate(DefaultPooledObject.java:196)
	at org.apache.tomcat.dbcp.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:449)
	at org.apache.tomcat.dbcp.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:360)
	at org.apache.tomcat.dbcp.dbcp2.PoolingDataSource.getConnection(PoolingDataSource.java:118)
	at org.apache.tomcat.dbcp.dbcp2.BasicDataSource.getConnection(BasicDataSource.java:1412)
	at com.dotmarketing.db.DbConnectionFactory.getConnection(DbConne
```

**Comment by wezell** [comment]
Yup - we need the finally{ db.close()} logic here.  

https://github.com/dotCMS/core/blob/master-3.6/src/com/dotmarketing/quartz/job/FreeServerFromClusterJob.java#L69

@dsilvam - let's make sure that everyone knows that any thread or quartz job that is spun up needs to explicitly commit/close DB connections.

**Comment by dsilvam** [comment]
I see, we missed the explicit db close on that quartz. I am opening an issue for it

**Comment by dsilvam** [comment]
@wezell yes absolutely!

**Comment by wezell** [comment]
This is fixed #9159
